crypto/rand/rand_lib.c

```diff
@@ -46,7 +46,7 @@ void rand_read_tsc(RAND_poll_fn cb, void *arg)
     unsigned char c;
     int i;
 
-    for (i = 0; i < 10; i++) {
+    for (i = 0; i < TSC_READ_COUNT; i++) {
         c = (unsigned char)(OPENSSL_rdtsc() & 0xFF);
```

Comment:
Speaking of which. Two things. `& 0xFF` kind of suggests that you expect counter to increment with 1 cycle granularity. This is not true. Secondly, as for TSC_READ_COUNT. One can argue that first reading is kind of random, because it's likely to be triggered by external even like an arrival of datagram. But it's not at all given that specifically least significant bits would be "most random", because interrupts are risen at lower rate. Well, there *is* uncertainty in lower bits, yet it's not given that the least significant bits are "most random". One can even argue that second TSC reading bears certain randomness as an "echo" of uncertainty in how caches *were* populated at *first* call to callback, but beyond second reading timestamps would be pretty predictable(\*).

(\*) Well, there is apparent uncertainty in interaction with external memory, as discussed in OPENSSL_instrument_bus.pod (there also are some pictures at https://www.openssl.org/~appro/OPENSSL_instrument_bus/), but this is not something one can rely upon in just few samples.